src/highlight/highlightUtil.ts

```diff
@@ -58,3 +58,65 @@ export const centerHighlight = (shapes: Shape[]): Position => {
 export function isHighlight(annotation: Annotation): annotation is AnnotationHighlight {
     return annotation?.target?.type === Type.highlight;
 }
+
+export const dedupeRects = (rects: Shape[], threshold = 0.1): Shape[] => {
+    const dedupedRects: Shape[] = [];
+
+    rects.forEach(curr => {
+        const prev = dedupedRects.pop();
+        // empty list, push current
+        if (!prev) {
+            dedupedRects.push(curr);
+            return;
```

Comment:
We generally try to avoid multiple returns (outside of guard clauses) because it can make the function more difficult to step through.

Reply:
The `pop()` may return `undefined`, so this is a guard clauses. I removed an unnecessary `return` below.